### PR TITLE
Update prettier, use `@theguild/prettier-config`, add `Lint Prettier` CI action

### PR DESCRIPTION
**Pull request by dimaMachina**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,26 +6,13 @@
...
-    "./examples/graphql-helix-auth0",
-    "./examples/lambda-aws",
-    "./examples/azure-functions",
-    "./examples/google-cloud-functions",
-    "./examples/cloudflare-workers",
-    "./examples/with-esm",
-    "./examples/typegraphql",
-    "./examples/nexus",
-    "./examples/express-graphql"
+    "./examples/*"
```
Comment (dimaMachina) [inline package.json]: can be simplified with `"./examples/*"`

File: package.json [hunk 2]
```diff
@@ -6,26 +6,13 @@
...
   ],
   "scripts": {
     "prepare": "husky install",
     "postinstall": "patch-package",
     "lint": "eslint --config .eslintrc.json --ext .ts .",
+    "lint:prettier": "prettier --ignore-path .gitignore --check .",
```
Comment (dimaMachina) [inline package.json:29]: I guess `lint:prettier` is a better name than `prettier:check`

File: README.md [hunk 3]
```diff
@@ -37,23 +37,21 @@ yarn add graphql @envelop/core
...
+})
 ```
 
 The result of `envelop` is a function that allows you to get everything you need for the GraphQL execution: `parse`, `validate`, `contextBuilder` and `execute`. Use that to run the client's GraphQL queries. Here's a pseudo-code example of how it should look like:
 
 ```ts
 const httpServer = createServer()
 
-httpServer.on('request', (req, res) => {
+httpServer.on('request', async (req, res) => {
```
Comment (dimaMachina) [inline README.md:54]: `async` was missing as in line 71 we use await

File: packages/plugins/response-cache/README.md [hunk 4]
```diff
@@ -230,10 +230,10 @@ const getEnveloped = envelop({
   plugins: [
     // ... other plugins ...
     useResponseCache({
-      shouldCacheResult = myCustomShouldCacheResult,
-    }),
-  ],
-});
+      shouldCacheResult: myCustomShouldCacheResult
```
Comment (dimaMachina) [inline packages/plugins/response-cache/README.md]: a typo I guess

File: package.json [hunk 5]
```diff
@@ -79,17 +67,13 @@
   "resolutions": {
     "@changesets/git": "1.3.1",
     "esbuild": "^0.14.0",
-    "prettier": "2.5.1",
```
Comment (dimaMachina) [inline package.json:82]: why to have resolutions for prettier?

File: package.json [hunk 6]
```diff
@@ -79,17 +67,13 @@
...
   },
   "lint-staged": {
     "packages/**/src/**/*.{ts,tsx}": [
       "eslint --fix"
     ],
-    "**/*.{ts,tsx,graphql,yml}": [
-      "prettier --write"
-    ],
-    "**/*.json": [
+    "**/*": [
```
Comment (dimaMachina) [inline package.json]: to prettify all files, even markdown

File: packages/plugins/response-cache/README.md [hunk 7]
```diff
@@ -81,28 +81,28 @@ In order to use the Redis cache, you need to:
...
 const redis = new Redis({
   host: 'my-redis-db.example.com',
   port: '30652',
-  password: '1234567890',
-});
+  password: '1234567890'
+})
 
-const redis = new Redis("rediss://:1234567890@my-redis-db.example.com:30652");
+const redis = new Redis('redis://:1234567890@my-redis-db.example.com:30652')
```
Comment (B2o5T) [inline packages/plugins/response-cache/README.md:96]: typo

File: packages/plugins/response-cache/README.md [hunk 8]
```diff
@@ -160,55 +160,55 @@ const getEnveloped = envelop({
...
+})
 ```
 
 ### Customize if result should be cached
 
 You can define a custom function used to check if a query operation execution result should be cached.
 
 ```ts
-type ShouldCacheResultFunction = (params: { result: ExecutionResult }) => Boolean;
+type ShouldCacheResultFunction = (params: { result: ExecutionResult }) => Boolean
```
Comment (dimaMachina) [inline packages/plugins/response-cache/README.md]: i guess it's better
```suggestion
type ShouldCacheResultFunction = (params: { result: ExecutionResult }) => boolean
```

File: website/docs/core.mdx [hunk 9]
```diff
@@ -199,17 +199,17 @@ const getEnveloped = envelop({
 This utility is helpful when you want to enable a plugin only when a certain condition is met.
 
 ```ts
-import { envelop, useMaskedErrors, enableIf } from '@envelop/core';
+import { envelop, useMaskedErrors, enableIf } from '@envelop/core'
 
-const isProd = process.env === 'production';
+const isProd = process.env === 'production'
```
Comment (dimaMachina) [inline website/docs/core.mdx]: also typo?
```suggestion
const isProd = process.env.NODE_ENV === 'production'
```

File: packages/plugins/response-cache/README.md (hunk 7 above)
Comment (dimaMachina) [inline packages/plugins/response-cache/README.md]:
```suggestion
const redis = new Redis('rediss://:1234567890@my-redis-db.example.com:30652')
```

File: package.json (hunk 1 above)
Comment (n1ru4l) [reply package.json]: No, it cannot as some plugins are not part of the workspace (they use the experimental defer & stream). Making them part of the workspace breaks them.

Comment (n1ru4l) [inline package.json]: please revert this change as we have some examples that are not part of the workspace that break due to dependency hoisting.
